routing/index_graph_starter.cpp

```diff
@@ -4,186 +4,96 @@
 
 namespace routing
 {
-IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, RoadPoint const & startPoint,
-                                     RoadPoint const & finishPoint)
-  : m_graph(graph)
-  , m_start(graph, startPoint, graph.GetNumJoints())
-  , m_finish(graph, finishPoint, graph.GetNumJoints() + 1)
+IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, FakeVertex const & start,
+                                     FakeVertex const & finish)
+  : m_graph(graph), m_start(start, false /* isFinish */), m_finish(finish, true /* isFinish */)
 {
-  m_start.SetupJointId(graph);
-
-  if (startPoint == finishPoint)
-    m_finish.m_jointId = m_start.m_jointId;
-  else
-    m_finish.SetupJointId(graph);
 }
 
-m2::PointD const & IndexGraphStarter::GetPoint(Joint::Id jointId)
+m2::PointD const & IndexGraphStarter::GetPoint(Segment const & segment, bool isOutgoing)
 {
-  if (jointId == m_start.m_fakeId)
-    return m_graph.GetGeometry().GetPoint(m_start.m_point);
+  if (segment == m_start.GetFakeSegment())
+    return m_start.GetPoint();
 
-  if (jointId == m_finish.m_fakeId)
-    return m_graph.GetGeometry().GetPoint(m_finish.m_point);
+  if (segment == m_finish.GetFakeSegment())
+    return m_finish.GetPoint();
 
-  return m_graph.GetPoint(jointId);
+  return m_graph.GetGeometry().GetPoint(segment.GetRoadPoint(isOutgoing));
 }
 
-void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
-                                     vector<RoutePoint> & routePoints)
+size_t IndexGraphStarter::GetRouteNumPoints(vector<Segment> const & route) const
 {
-  if (route.size() < 2)
-  {
-    if (route.size() == 1)
-      routePoints.emplace_back(m_start.m_point, 0.0 /* time */);
-    return;
-  }
-
-  routePoints.reserve(route.size() * 2);
+  // if route contains start and finish fakes only, it doesn't have any points.
```

Comment:
Really? But it still has start and finish points, why do we need this complex logic?